cmd/start.go

```diff
@@ -43,21 +61,35 @@ func initStartCommand(commonFlags []cli.Flag) *cli.Command {
 			ctx := context.Background()
 			logger := logs.InitLogger(c.Bool("debug"))
 
+			// Handle '~' exapansion for config files
+			hopsPath, err := homedir.Expand(c.String("hops"))
+			if err != nil {
+				return err
+			}
+			kubeconfigPath, err := homedir.Expand(c.String("kubeconfig"))
+			if err != nil {
+				return err
+			}
+			keyfilePath, err := homedir.Expand(c.String("keyfile"))
+			if err != nil {
+				return err
+			}
+
 			hopsServer := &hops.HopsServer{
 				Console: hops.Console{
 					Address: c.String("address"),
 					Serve:   c.Bool("serve-console"),
 				},
-				HopsPath: c.String("hops"),
+				HopsPath: hopsPath,
```

Comment:
This can be made DRY fairly easily if we use flag actions:

```go
func expandHomePath(flagName string) func(c *cli.Context, val string) error {
	return func(c *cli.Context, val string) error {
		expandedPath, err := homedir.Expand(c.String(flagName))
		if err != nil {
			return err
		}

		c.Set(flagName, expandedPath)
		return nil
	}
}

// Use it in a flag like so:
&cli.StringFlag{
	Name:     "keyfile",
	Usage:    "Path to the hiphops key",
	Value:    defaultKeyFilePath,
	Category: commonFlagCategory,
	Action: expandHomePath("keyfile"),
}
```

Reply:
Also, I believe this code makes lines 57-59 irrelevant in `cmd.go`, since we'd instead set a default value on each flag `Value: "~/whatever.file"